templates/vue/src/store/mutations.js

```diff
@@ -78,3 +78,36 @@ export function updateEntry(state, { answerType, entry, nodeId, questionId }) {
   entries[answerType] = Object.values(entry)[0]
   question.entries = entries
 }
+
+export function updateTydeProgress(state, {parentId, isParentModule}){
+  const parentNode = state.nodes[Helpers.findNodeIndex(parentId, state)]
+  const childNodeIds = getChildIds(state, parentId)
+  const childNodes = childNodeIds.map(id => state.nodes[Helpers.findNodeIndex(id, state)])
+  if(isParentModule){ // parentNode is a module, and all children are stages
+    const childProgress = childNodes.map(stage => stage.tydeProgress)
+    const reducer = (accumulator, progress) => accumulator + progress
+    var curProgress = childProgress.reduce(reducer, 0) / childNodes.length
+  } else { // parentNode is a stage, and all children are nodes (topics)
+    var childProgress = childNodes.map(topic => topic.completed)
+    const reducer = (accumulator, completed) => accumulator + (completed === true ? 1 : 0)
+    var curProgress = childProgress.reduce(reducer, 0) / childNodes.length
```

Comment:
This line seems to be repeated in the `if` and `else` statements. Please move it out of that conditional. You actually don't even need to assign it to `curProgress`, rather directly assign it to `parentNode.tydeProgress` below.

Reply:
You forgot to move it out of the if/else statement, but no worries - I will make this change and commit it before merging this in.